### PR TITLE
feat(mls): adopt commit bundles (AR-2507)

### PR DESCRIPTION
**Pull request by mchenani**
----
#### PR Submission Checklist for internal contributors

- The **PR Title**
  - [x] conforms to the style of semantic commits messages¹ supported in Wire's Github Workflow²
  - [x] contains a reference JIRA issue number like `SQPIT-764`
  - [x] answers the question: _If merged, this PR will: ..._ ³

- The **PR Description**
  - [ ] is free of optional paragraphs and you have filled the relevant parts to the best of your ability
----

# What's new in this PR?
Using commit bundles instead of sending Welcome and Commit messages separately. 
### Issues

Adopt the new approach of commit-bundles.

### Causes (Optional)

Adopt changes.

### Solutions

Send the WelcomeMessage, CommitMessage and the GroupState via commit-bundles and remove the separated calls.

Needs releases with:

- [ ] GitHub link to other pull request

#### Test Coverage (Optional)
WIP
- [ ] I have added automated test to this contribution

#### How to Test

Only works on API-V3: To test you need to run the AndroidApp against the Staging and having the Api development enabled.

----
#### PR Post Submission Checklist for internal contributors (Optional)

 - [ ] Wire's Github Workflow has automatically linked the PR to a JIRA issue
----
#### PR Post Merge Checklist for internal contributors

 - [ ] If any soft of configuration variable was introduced by this PR, it has been added to the relevant documents and the CI jobs have been updated.
----
##### References
1. https://sparkbox.com/foundry/semantic_commit_messages
1. https://github.com/wireapp/.github#usage
1. E.g. `feat(conversation-list): Sort conversations by most emojis in the title #SQPIT-764`.

### CODE REVIEW

File: logic/src/commonTest/kotlin/com/wire/kalium/logic/data/conversation/MLSConversationRepositoryTest.kt [hunk 1]
```diff
@@ -232,9 +225,7 @@ class MLSConversationRepositoryTest {
             .withClaimKeyPackagesSuccessful()
             .withGetMLSClientSuccessful()
             .withAddMLSMemberSuccessful()
-            .withSendWelcomeMessageSuccessful()
-            .withSendMLSMessageSuccessful()
-            .withCommitAcceptedSuccessful()
```
Comment (typfel) [inline logic/src/commonTest/kotlin/com/wire/kalium/logic/data/conversation/MLSConversationRepositoryTest.kt:237]: Commit accepted also removed here.

File: logic/src/commonTest/kotlin/com/wire/kalium/logic/data/conversation/MLSConversationRepositoryTest.kt [hunk 2]
```diff
@@ -496,11 +485,9 @@ class MLSConversationRepositoryTest {
             .withCommitPendingProposalsSuccessful()
             .withGetMLSClientSuccessful()
             .withRemoveMemberSuccessful()
-            .withSendMLSMessageSuccessful()
+            .withCommitBundleSuccessful()
             .withUpdateConversationGroupStateSuccessful()
             .withFetchClientsOfUsersSuccessful()
-            .withCommitAcceptedSuccessful()
```
Comment (typfel) [inline logic/src/commonTest/kotlin/com/wire/kalium/logic/data/conversation/MLSConversationRepositoryTest.kt:502]: Commit accepted also removed here.

File: logic/src/commonTest/kotlin/com/wire/kalium/logic/data/conversation/MLSConversationRepositoryTest.kt [hunk 3]
```diff
@@ -473,11 +464,9 @@ class MLSConversationRepositoryTest {
             .withCommitPendingProposalsReturningNothing()
             .withGetMLSClientSuccessful()
             .withRemoveMemberSuccessful()
-            .withSendMLSMessageSuccessful(events = listOf(Arrangement.MEMBER_LEAVE_EVENT))
+            .withCommitBundleSuccessful(events = listOf(Arrangement.MEMBER_LEAVE_EVENT))
             .withUpdateConversationGroupStateSuccessful()
             .withFetchClientsOfUsersSuccessful()
-            .withCommitAcceptedSuccessful()
```
Comment (typfel) [inline logic/src/commonTest/kotlin/com/wire/kalium/logic/data/conversation/MLSConversationRepositoryTest.kt:479]: Commit accepted also removed here.

File: logic/src/commonTest/kotlin/com/wire/kalium/logic/data/conversation/MLSConversationRepositoryTest.kt [hunk 4]
```diff
@@ -444,11 +438,9 @@ class MLSConversationRepositoryTest {
             .withCommitPendingProposalsReturningNothing()
             .withGetMLSClientSuccessful()
             .withRemoveMemberSuccessful()
-            .withSendMLSMessageSuccessful()
+            .withCommitBundleSuccessful()
             .withUpdateConversationGroupStateSuccessful()
             .withFetchClientsOfUsersSuccessful()
-            .withCommitAcceptedSuccessful()
```
Comment (typfel) [inline logic/src/commonTest/kotlin/com/wire/kalium/logic/data/conversation/MLSConversationRepositoryTest.kt:450]: Commit accepted also removed here.

**Review by typfel — CHANGES_REQUESTED** [review]

File: logic/src/commonTest/kotlin/com/wire/kalium/logic/data/conversation/MLSConversationRepositoryTest.kt [hunk 5]
```diff
@@ -69,9 +69,7 @@ class MLSConversationRepositoryTest {
             .withGetPublicKeysSuccessful()
             .withCreateMLSConversationSuccessful()
             .withAddMLSMemberSuccessful()
-            .withSendWelcomeMessageSuccessful()
-            .withSendMLSMessageSuccessful()
-            .withCommitAcceptedSuccessful()
```
[the remark this answers is not in the document]
Comment (mchenani) [reply logic/src/commonTest/kotlin/com/wire/kalium/logic/data/conversation/MLSConversationRepositoryTest.kt:74]: TBH, I think that would be extra step after sending the commit and in this test we don't need that step [only we check the main sequence and we don't wait for the commit to be accepted in this function]
But we can sync online, I feel that I'm missing something maybe.

[the remark this answers is not in the document]
Comment (typfel) [reply logic/src/commonTest/kotlin/com/wire/kalium/logic/data/conversation/MLSConversationRepositoryTest.kt:74]: The commit is accepted I see now that we have a

`withCommitAcceptedSuccessful()` and `withSendCommitAcceptedSuccessful()` which I guess is a bit confusing where `withCommitAcceptedSuccessful()` is group of `with()` including `withCommitAcceptedSuccessful()`

Maybe should remove `withCommitAcceptedSuccessful()` or rename it

File: logic/src/commonTest/kotlin/com/wire/kalium/logic/data/conversation/MLSConversationRepositoryTest.kt [hunk 6]
```diff
@@ -819,8 +825,9 @@ class MLSConversationRepositoryTest {
                 .then { NetworkResponse.Success(SendMLSMessageResponse(TIME, events), emptyMap(), 201) }
         }
 
-        fun withCommitBundleSuccessful() = apply {
-            withSendMLSMessageSuccessful()
+        fun withCommitBundleSuccessful(events: List<EventContentDTO> = emptyList()) = apply {
+            withSendMLSMessageSuccessful(events)
```
Comment (typfel) [inline logic/src/commonTest/kotlin/com/wire/kalium/logic/data/conversation/MLSConversationRepositoryTest.kt:829]: This should also be removed I guess but then again I guess we can remove this whole function.

**Review by typfel — APPROVED** [review]

**Pull request merged**